src/widgets.tsx

```diff
@@ -550,10 +1022,88 @@ export const ProfileImage = (props:{style?:ImageStyle}):JSX.Element => {
 
     return (
         <>
-            {userProfileImage === undefined ?
+            { userProfileImage === undefined ?
                 <Image source={DEFAULT_PROFILE_ICON} style={styles.profileImage}></Image> : <Image source={{uri: userProfileImage}} style={styles.profileImage}></Image>
             }
+
         </>
 
     );
 }
+
+export const RequestorProfileImage = (props:{style?:ImageStyle, imageUri?:string, userID?:number}):JSX.Element => {
+    const userID = useAppSelector((state: RootState) => state.account.userID);
+    const userProfileImage = useAppSelector((state: RootState) => state.account.userProfile.image);
+    const jwt = useAppSelector((state: RootState) => state.account.jwt);
+
+    const DEFAULT_PROFILE_ICON = require("../assets/profile-icon-blue.png");
+
+    const [requestorImage, setRequestorImage] = useState<ImageSourcePropType>(DEFAULT_PROFILE_ICON);
+
+    const RequestAccountHeader = {
+        headers: {
+          "jwt": jwt, 
+        }
+      }
+
+    const styles = StyleSheet.create({
+        profileImage: {
+            height: 100,
+            width: 100,
+            borderRadius: 15,
+            alignSelf: "center",
+            ...props.style
+        },
+    })
+
+    const fetchProfileImage = async () => {
+        await axios.get(`${DOMAIN}/api/user/` + props.userID + '/image', RequestAccountHeader).then(response => {
+            setRequestorImage({uri: response.data})
+        }).catch((error:AxiosError) => {console.log(error)})
+    }
+
+    useEffect(() => {
+        if (props.imageUri !== undefined) setRequestorImage({uri: props.imageUri})
+        else if (props.userID !== undefined) {
+            if (props.userID == userID) setRequestorImage({uri: userProfileImage})
+            else fetchProfileImage();
+        }
+    },[])
+    return <Image source={requestorImage} style={styles.profileImage}></Image> 
```

Comment:
Can this return `<ProfileImage />` ?

Reply:
The ProfileImage prop is exclusive to the currently logged in user. Because of that, it uses an AppSelector from redux to display the profile image. It doesn't need a src prop because everything this component needs can be fetched locally.

The Image tag above is in the RequestorProfileImage component, This component is for rendering profile images for users that are not guaranteed to be the currently logged in user. Because of that, it has different logic.

Reply:
Gotcha, good design!